dbms/CMakeLists.txt

```diff
@@ -564,11 +564,14 @@ if (USE_JEMALLOC)
     dbms_target_include_directories (SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR}) # used in Interpreters/AsynchronousMetrics.cpp
     target_include_directories (clickhouse_new_delete SYSTEM BEFORE PRIVATE ${JEMALLOC_INCLUDE_DIR})
 
-    if(NOT MAKE_STATIC_LIBRARIES AND ${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$")
-        # mallctl in dbms/src/Interpreters/AsynchronousMetrics.cpp
-        # Actually we link JEMALLOC to almost all libraries.
-        # This is just hotfix for some uninvestigated problem.
-        target_link_libraries(clickhouse_interpreters PRIVATE ${JEMALLOC_LIBRARIES})
+    if (NOT MAKE_STATIC_LIBRARIES)
+        target_link_libraries(clickhouse_common_io PRIVATE ${JEMALLOC_LIBRARIES})
```

Comment:
Don't understand the details.

Why we don't check that we are using JeMalloc?
Why it's linked to `clickhouse_interpreters` conditionally and to `clickhouse_common_io` unconditionally?
Why the condition `${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$"` looks so strage?
How it can be possible that the JeMalloc library is static despite the `NOT MAKE_STATIC_LIBRARIES`?

Need to provide comments for all these questions and also need to provide a reproducible example (it can be clickhouse build in Docker).

Note that we have checks for shared build in CI and it is green.

Reply:
(*This is not mine PR, but I remember when this had been introduced.*)

>Why we don't check that we are using JeMalloc?

There is `USE_JEMALLOC` above

>Why it's linked to clickhouse_interpreters conditionally and to clickhouse_common_io unconditionally?

It should be under `NOT MAKE_STATIC_LIBRARIES` but not `${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$`

>Why the condition ${JEMALLOC_LIBRARIES} MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$" looks so strage?

0006f2a370deccf5cd7af0d6e6a75e5351c5c626

>How it can be possible that the JeMalloc library is static despite the NOT MAKE_STATIC_LIBRARIES?

`-DUNBUNLDED=ON -DENABLE_JEMALLOC=ON` (will use system-wide jemalloc, that can be shared)

>Need to provide comments for all these questions and also need to provide a reproducible example (it can be clickhouse build in Docker).

How about removing it all together? https://github.com/ClickHouse/ClickHouse/pull/9006#discussion_r379909514

Reply:
> How about removing it all together? #9006 (comment)

Much better!

Let's make another PR where it is removed and close this PR.